### PR TITLE
Make String parseValue coercion consistent with JS implementation & Gradle JCenter fix

### LINKED ISSUE #2990: Fix GraphQL String coercion
String parseValue should only accept Strings, nothing els.e

### PR DESCRIPTION
**Pull request by dondonz**
This PR makes String `parseValue` coercion consistent with the JS reference implementation, by only allowing String input values. Fixes #2990.

Consistent implementation of String coercion is particularly important for [String-based custom scalars](https://scalars.graphql.org/template-string.html).

Note that this is a breaking change.

`parseValue` in JS implementation: https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L156

----
This PR also fixes the Gradle settings to avoid redirecting to the defunct JCenter when the Gradle module metadata is not published by a plugin (in our case, the jmh plugin). https://github.com/gradle/gradle/issues/22864#issuecomment-1327978524

There was a problem with a JCenter SSL certificate expiring earlier today.

### CODE REVIEW

File: src/test/groovy/graphql/ScalarsStringTest.groovy [hunk 1]
```diff
@@ -54,7 +55,6 @@ class ScalarsStringTest extends Specification {
     def "String serialize #value into #result (#result.class)"() {
         expect:
         Scalars.GraphQLString.getCoercing().serialize(value, GraphQLContext.default, Locale.default) == result
-        Scalars.GraphQLString.getCoercing().parseValue(value, GraphQLContext.default, Locale.default) == result
```
Comment (dondonz) [inline src/test/groovy/graphql/ScalarsStringTest.groovy:57]: Splitting out tests for `serialize` and `parseValue`

File: settings.gradle [hunk 2]
```diff
@@ -1,2 +1,16 @@
-rootProject.name = 'graphql-java'
+pluginManagement {
+    repositories {
+        mavenCentral()
+        maven {
```
Comment (dondonz) [inline settings.gradle:4]: Snippet from https://github.com/gradle/gradle/issues/22864#issuecomment-1327978524

**Comment by federicorispo** [comment]
@dondonz Since this is a breaking change I would consider to align also the other types:
 - GraphqlBooleanCoercing.java ([graphql-js implementation](https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L194))
 - GraphqlIDCoercing.java ([graphql-js implementation](https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L233))
 - GraphqlFloatCoercing.java ([graphql-js implementation](https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L112))
 - GraphqlIntCoercing.java ([graphql-js implementation](https://github.com/graphql/graphql-js/blob/main/src/type/scalars.ts#L55))
 
 I can help with these if you like

**Review by bbakerman — APPROVED** [review]

**Comment by dondonz** [comment]
> @dondonz Since this is a breaking change I would consider to align also the other types:

Thanks, that's a good pickup @federicorispo. I'll add the other `parseValue` changes in the next PR

**Pull request merged**